### PR TITLE
Square root and Power primitives added

### PR DESCRIPTION
**Pull request by parsa**
Part of issue #2

### CODE REVIEW

File: src/execution_tree/primitives/power_operation.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            using operand_type = ir::node_data<double>;
+            using operands_type = std::vector<operand_type>;
+
+            primitive_result_type power0d(operands_type && ops) const
+            {
+                ops[0][0] = std::pow(ops[0][0], ops[1][0]);
+                return std::move(ops[0]);
+            }
+
+            // lhs_num_dims == 1
```
Comment (hkaiser) [inline src/execution_tree/primitives/power_operation.cpp]: This comment is misleading

File: src/execution_tree/primitives/power_operation.cpp [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            primitive_result_type powerxd(operands_type && ops) const
+            {
+                operand_type& lhs = ops[0];
+                operand_type& rhs = ops[1];
+
+                lhs.matrix() = blaze::pow(lhs.matrix(), rhs[0]);
```
Comment (hkaiser) [inline src/execution_tree/primitives/power_operation.cpp]: Should we add an exception for the case if the rhs is not a scalar?

File: src/execution_tree/primitives/square_root_operation.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+
+    ///////////////////////////////////////////////////////////////////////////////
+    namespace phylanx {
+    namespace execution_tree {
+        namespace primitives
+        {
+            ///////////////////////////////////////////////////////////////////////////
+            std::vector<match_pattern_type> const square_root_operation::match_data =
+            {
+                hpx::util::make_tuple("square_root", "square_root(_1, _2)", &create<square_root_operation>)
```
Comment (hkaiser) [inline src/execution_tree/primitives/square_root_operation.cpp]: Please avoid long lines.

File: src/execution_tree/primitives/power_operation.cpp (hunk 2 above)
Comment (parsa) [reply src/execution_tree/primitives/power_operation.cpp]: No. `rhs` is guaranteed to be a scalar when the code reaches here. See `eval()` (line 97)

File: src/execution_tree/primitives/power_operation.cpp (hunk 1 above)
Comment (parsa) [reply src/execution_tree/primitives/power_operation.cpp]: Removed.

**Pull request merged**